pkg/util/ready/ready.go

```diff
@@ -24,6 +27,26 @@ import (
 	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
 )
 
+// IsMatchingSyscallError returns true when the error is one of the Errno's in match
+// it deals with the different ways of wrapping the syscalls.
+func IsMatchingSyscallError(err error, match []syscall.Errno) bool {
+	switch t := err.(type) {
+	case *url.Error:
+		return IsMatchingSyscallError(t.Err, match)
```

Comment:
I believe golang doesn't support tail recursion - use a loop here instead.  I know this is a bit of a nit, but recursing here might encourage others to do it in a place where it does matter.

Reply:
ok, i'll change to a for loop